app/slackimport.go

```diff
@@ -656,6 +662,7 @@ func (a *App) SlackImport(fileData multipart.File, fileSize int64, teamID string
 	}
 
 	var channels []SlackChannel
+	var tempChannels []SlackChannel
```

Comment:
I think it would be a bit neater to have:

```suggestion
	var publicChannels []SlackChannel
	var privateChannels []SlackChann
	var groupChannels []SlackChannelel
	var directChannels []SlackChannel
```

And then in the if block below, just assign each of these as appropriate, and then after the if block, have:

```
channels := append(publicChannels, privateChannels, GroupChannels, directChannels)

Reply:
Good idea, but the channels append doesn't work, because I can't append multiple slices like that. 
So I'd either have to create a slice of slices and then iterate after over that to append to the channels slice or do an append for each channel individually.

I've commited the solution with iterating over the channel slices, it's a bit more lines of code than before, but definitely looks cleaner now.